src/Libraries/DSCPython/DSCPythonCodeCompletionProviderCore.cs

```diff
@@ -0,0 +1,1292 @@
+            BasicVariableTypes.Add(Tuple.Create(LIST_VARIABLE, typeof(PyList)));
+            BasicVariableTypes.Add(Tuple.Create(DICT_VARIABLE, typeof(PyDict)));
+            
+            Python.Included.Installer.SetupPython().Wait();
+            if (!PythonEngine.IsInitialized)
+            {
+                PythonEngine.Initialize();
+                PythonEngine.BeginAllowThreads();
+            }
+
+            IntPtr gs = PythonEngine.AcquireLock();
+            try
+            {
+                using (Py.GIL())
+                {
+                    if (Scope == null)
+                    {
+                        Scope = CreateGlobalScope();
+                    }
+
+                    var assemblies = AppDomain.CurrentDomain.GetAssemblies();
+
+                    // Determine if the Revit API is available in the given context
+                    if (assemblies.Any(x => x.GetName().Name == "RevitAPI"))
+                    {
+                        // Try to load Revit Type for autocomplete
+                        try
+                        {
+                            var revitImports =
+                                "import clr\nclr.AddReference('RevitAPI')\nclr.AddReference('RevitAPIUI')\nfrom Autodesk.Revit.DB import *\nimport Autodesk\n";
+
+                            Scope.Exec(revitImports);
+
+                            clrModules.Add("RevitAPI");
+                            clrModules.Add("RevitAPIUI");
+                        }
+                        catch
+                        {
+                            Log("Failed to load Revit types for autocomplete. Python autocomplete will not see Autodesk namespace types.");
+                        }
+                    }
+
+                    // Determine if the ProtoGeometry is available in the given context
+                    if (assemblies.Any(x => x.GetName().Name == "ProtoGeometry"))
+                    {
+                        // Try to load ProtoGeometry Type for autocomplete
+                        try
+                        {
+                            var libGImports =
+                                "import clr\nclr.AddReference('ProtoGeometry')\nfrom Autodesk.DesignScript.Geometry import *\n";
+
+                            Scope.Exec(libGImports);
+                            clrModules.Add("ProtoGeometry");
+                        }
+                        catch (Exception e)
+                        {
+                            Log(e.ToString());
+                            Log("Failed to load ProtoGeometry types for autocomplete. Python autocomplete will not see Autodesk namespace types.");
+                        }
+                    }
+                }
+            }
+            finally
+            {
+                PythonEngine.ReleaseLock(gs);
+            }
+        }
+
+        #endregion
+
+        private PyScope CreateGlobalScope()
```

Comment:
this function again is named the wrong thing for what it does.

Reply:
Changed the name to CreateUniquePythonScope()